validator/accounts/v2/accounts_deposit_test.go

```diff
@@ -0,0 +1,206 @@
+
+	// Test the case of providing the public keys via command-line.  We also pass in the test eth1 private key file.
+	// hexPubkeysString holds 1 less than all the accounts.
+	depositConfig, err = createDepositConfigHelper(t, &depositTestWalletConfig{
+		keymanagerKind:              v2keymanager.Derived,
+		walletDir:                   walletDir,
+		walletPasswordFile:          passwordFilePath,
+		skipDepositConfirmationFlag: true,
+		publicKeysFlag:              hexPubkeysString,
+		httpWeb3ProviderFlag:        "http://localhost:8545",
+		eth1PrivateKeyFile:          eth1PrivateKeyFile.Name(),
+	})
+	require.NoError(t, err)
+	require.Equal(t, 2, len(depositConfig.DepositPublicKeys), "wrong number of public keys")
+
+	// Compare the keys in the config object with the keys we obtained earlier from the keymanager
+	for keyNum, configPubKey := range depositConfig.DepositPublicKeys {
+		for index, eachByte := range bytesutil.ToBytes48(configPubKey.Marshal()) {
+			if eachByte != pubkeys[keyNum][index] {
+				require.NoError(t, errors.New("public keys do not match"))
+			}
+		}
+	}
+
+	// Now we test when private key file is not provided but rather the keystore and keystore password files
+	depositConfig, err = createDepositConfigHelper(t, &depositTestWalletConfig{
+		keymanagerKind:              v2keymanager.Derived,
+		walletDir:                   walletDir,
+		walletPasswordFile:          passwordFilePath,
+		skipDepositConfirmationFlag: true,
+		depositAllAccountsFlag:      true,
+		httpWeb3ProviderFlag:        "http://localhost:8545",
+		eth1KeystoreFile:            "This would be eth1 keystore file path",
+		eth1KeystorePasswordFile:    "This would be eth1 keystore password file path",
+	})
+	require.NoError(t, err)
+	require.Equal(t, 3, len(depositConfig.DepositPublicKeys), "wrong number of public keys")
+	require.Equal(t, "This would be eth1 keystore file path", depositConfig.Eth1KeystoreUTCFile,
+		"eth1 keystore file path incorrect")
+	require.Equal(t, "This would be eth1 keystore password file path", depositConfig.Eth1KeystorePasswordFile,
+		"eth1 keystore password file path incorrect")
+	require.Equal(t, "", depositConfig.Eth1PrivateKey, "eth1 private key should be empty string")
+}
+
+// createDepositConfigHelper returns a SendDepositConfig when given a particular wallet configuration.
+func createDepositConfigHelper(t *testing.T, config *depositTestWalletConfig) (*derived.SendDepositConfig, error) {
+	cliCtx := setupWalletCtxforDeposits(t, config)
+	wallet, err := OpenWalletOrElseCli(cliCtx, func(cliCtx *cli.Context) (*Wallet, error) {
+		err := errors.New("could not open wallet")
+		require.NoError(t, err)
+		return nil, err
+	})
+	require.NoError(t, err)
+
+	keymanager, err := wallet.InitializeKeymanager(
+		cliCtx.Context,
+		true, /* skip mnemonic confirm */
+	)
+	require.NoError(t, err)
+	km, ok := keymanager.(*derived.Keymanager)
+	if !ok {
```

Comment:
```suggestion
require.Equal(t, true, ok, "keymanager must be derived type")
```

Reply:
Thanks.  Will address in next commit soon.